students/ZhuojingXie/lesson09/mailroom5.py

```diff
@@ -0,0 +1,111 @@
+import sys
+import os
+from collections import defaultdict
+donor_data = defaultdict(list,{'Andy':[960,256,123.5,40],'Bryce':[30,45,27],
+                            'Charile': [25,50], 'David':[10], 'Elaine' :[75,26]})
+
+class Donor():
+
+
+    """this is Donor class to save donor's information"""
+    def __init__(self, name, donation):
+        self.name = name
+        self.donation = donation
+
+
+    def add_donor_in_data(self):
+        return donor_data[self.name].append(self.donation)
+
+
+class Collection:
+
+    def __init__(self, donor_data):
+        self.donor_data = donor_data
+
+    def all_donor(self):
+        temp_keys=[]
+        for keys in self.donor_data:
+            temp_keys.append(keys)
+        return temp_keys
```

Comment:
should  be `for key` since you are iterating over a single key, also use comprehension to avoid "temp" vars.. even then you don't need any loops: `returns list(self.donor_data)`